src/Nethermind/Nethermind.Serialization.Rlp/TxDecoder.cs

```diff
@@ -389,6 +423,46 @@ public void Encode(RlpStream stream, T? item, RlpBehaviors rlpBehaviors = RlpBeh
                 return;
             }
 
+            bool isSszTransaction = item.Type is TxType.Blob;
+            if (isSszTransaction)
+            {
+                byte[]? encodedTx;
+                int length;
+                if (forSigning)
+                {
+                    length = Ssz.Ssz.TransactionLength(item) + sizeof(TxType);
+                    encodedTx = ArrayPool<byte>.Shared.Rent(length);
+                    int offset = 1;
+                    Ssz.Ssz.Encode(encodedTx, item, ref offset);
+                }
+                else if ((rlpBehaviors & RlpBehaviors.InNetworkForm) == RlpBehaviors.InNetworkForm)
+                {
+                    length = Ssz.Ssz.BlobTransactionNetworkWrapperLength(item) + sizeof(TxType);
+                    encodedTx = ArrayPool<byte>.Shared.Rent(length);
+                    Ssz.Ssz.EncodeSignedWrapper(encodedTx.AsSpan(1), item);
+                }
+                else
+                {
+                    length = Ssz.Ssz.SignedBlobTransactionLength(item) + sizeof(TxType);
+                    encodedTx = ArrayPool<byte>.Shared.Rent(length);
+                    Ssz.Ssz.EncodeSigned(encodedTx.AsSpan(1), item);
+                }
+
+                encodedTx[0] = (byte)item.Type;
+
+                if ((rlpBehaviors & RlpBehaviors.SkipTypedWrapping) == RlpBehaviors.SkipTypedWrapping)
+                {
+                    stream.Write(encodedTx.AsSpan(0, length));
+                }
+                else
+                {
+                    stream.Encode(encodedTx.AsSpan(0, length));
```

Comment:
I'm not sure about correctness of that part (`if(SkipTypedWrapping) Write else Encode`). Could you explain the logic behind it?

Reply:
SkipTypedWrapping flag means we do not encode data as an RLP array (Encode), but just pass it as is (Write). It's the same as this logic below:
```
if (item.Type != TxType.Legacy)
{
    if ((rlpBehaviors & RlpBehaviors.SkipTypedWrapping) == RlpBehaviors.None)
    {
        stream.StartByteArray(sequenceLength + 1, false);
    }

    stream.WriteByte((byte)item.Type);
}
```

Reply:
Thanks for the clarification!